### DISCUSSION TITLE
How to set api key for posting events.

### DISCUSSION THREAD

**Discussion by toshanmugaraj** [plausible/analytics#3381] [Self-Hosted Support]
Right now I don't see option to set api key from client side to post analytics events. If there is no such mechanism, anyone can dumb to the analytics server.

**Comment by ruslandoga** [comment]
👋 @toshanmugaraj 

I think you've already asked about this: https://github.com/plausible/analytics/discussions/3331

And the answer is the same: adding the API key to the client requests from the browser makes it visible and doesn't prevent anyone from copying it and using it for bogus requests.
